tests/test_sequence_interpolator.py

```diff
@@ -0,0 +1,429 @@
+        ), "Nearest neighbor interpolation does not match expected data"
+        assert valid.shape == (10,), f"Expected valid.shape == (10,), got {valid.shape}"
+        assert interp.shape == (
+            10,
+            n_signals,
+        ), f"Expected interp.shape == (10, {n_signals}), got {interp.shape}"
+
+
+@pytest.mark.parametrize("n_signals", [0, 1, 10, 50])
+@pytest.mark.parametrize("sampling_rate", [3.0, 10.0, 100.0])
+def test_nearest_neighbor_interpolation_with_inbetween_times(n_signals, sampling_rate):
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=n_signals,
+            use_mem_mapped=True,
+            t_end=5.0,
+            sampling_rate=sampling_rate,
+        )
+    ) as (timestamps, data, _, seq_interp):
+        assert isinstance(
+            seq_interp, SequenceInterpolator
+        ), "Interpolation object is not a SequenceInterpolator"
+
+        delta_t = 1.0 / sampling_rate
+
+        # timestamps multiplied by 0.8 should be floored to the same timestamp
+        times = timestamps[:10] + 0.8 * delta_t
+        interp, valid = seq_interp.interpolate(times=times)
+        assert times.shape == valid.shape, "All samples should be valid"
+        assert np.allclose(
+            interp, data[:10]
+        ), "Nearest neighbor interpolation does not match expected data"
+
+        # timestamps multiplied by 1.2 should be floored to the next timestamp
+        times = timestamps[:10] + 1.2 * delta_t
+        interp, valid = seq_interp.interpolate(times=times)
+        assert times.shape == valid.shape, "All samples should be valid"
+        assert np.allclose(
+            interp, data[1:11]
+        ), "Nearest neighbor interpolation does not match expected data"
+
+
+@pytest.mark.parametrize("n_signals", [0, 1, 10, 50])
+@pytest.mark.parametrize("sampling_rate", [3.0, 10.0, 100.0])
+@pytest.mark.parametrize("use_mem_mapped", [False, True])
+def test_nearest_neighbor_interpolation_with_phase_shifts(
+    n_signals, sampling_rate, use_mem_mapped
+):
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=n_signals,
+            use_mem_mapped=use_mem_mapped,
+            t_end=5.0,
+            sampling_rate=sampling_rate,
+            shifts_per_signal=True,
+        )
+    ) as (timestamps, data, shift, seq_interp):
+        assert isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Interpolation object is not a PhaseShiftedSequenceInterpolator"
+
+        delta_t = 1.0 / sampling_rate
+        times = (
+            timestamps[1:11] + 1e-9
+        )  # Add a small epsilon to avoid floating point errors
+        interp, valid = seq_interp.interpolate(times=times)
+        assert times.shape == valid.shape, "All samples should be valid"
+        assert np.allclose(
+            interp, data[0:10]
+        ), "Nearest neighbor interpolation does not match expected data"
```

Comment:
again here - for knn interpolation it should be exact equality and not allclose

plus this test basically says that we need more generations for phase_shifts as the whole point is that they change the indexes of the data returned (eg for neuron 1 we return indexes 0 to 10 and for neuron 2 we return indexes from 1 to 11 because neuron 1 has a bigger timeshift than neuron 2)

But I am fine to add these tests later (@tomschewski - the explanation above might be also helpful for you )

Reply:
@reneburghardt should we resolve this one as well, it seems to be fixed now, isn't it?